libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -729,6 +731,68 @@ impl Greenlight {
         Ok((new_state, payments))
     }
 
+    /// If there are any pending transactions supplied, this function checks
+    /// whether there was an update for that pending transaction in the current
+    /// sync round. If not, it checks whether the transaction exists on the
+    /// remote node. If it doesn't exist, the pending transaction is marked to
+    /// be deleted.
+    async fn update_pending_transactions(
+        &self,
+        payments: &[Payment],
+        pending: Vec<String>,
+    ) -> NodeResult<Vec<String>> {
+        if pending.is_empty() {
+            return Ok(Vec::new());
+        }
+
+        let mut pending_map: HashSet<String> = HashSet::from_iter(pending);
+        for payment in payments {
+            if pending_map.is_empty() {
+                return Ok(Vec::new());
+            }
+
+            // If the payment is already in the update, the pending payment will
+            // be resolved automatically.
+            pending_map.remove(&payment.id);
+        }
+
+        if pending_map.is_empty() {
+            return Ok(Vec::new());
+        }
+
+        let client = self.get_node_client().await?;
+
+        // If there's any pending payments remaining, check whether they exist
+        // in parallel.
+        let mut futures = Vec::new();
+        for hash in pending_map {
```

Comment:
Do you think it is better to initiate one query for all pending payments here instead a query per payment?

Reply:
You mean fetching all payments and checking whether our pending payment is there?
I don't think so, no, because usually this will be a single payment.